app/components/emoji/index.js

```diff
@@ -4,12 +4,40 @@
 import {connect} from 'react-redux';
 
 import {getCustomEmojisByName} from 'mattermost-redux/selectors/entities/emojis';
+import {getCurrentUserId} from 'mattermost-redux/selectors/entities/users';
+import {getConfig} from 'mattermost-redux/selectors/entities/general';
+import {Client4} from 'mattermost-redux/client';
+import {isMinimumServerVersion} from 'mattermost-redux/utils/helpers';
+
+import {EmojiIndicesByAlias, Emojis} from 'app/utils/emojis';
 
 import Emoji from './emoji';
 
-function mapStateToProps(state) {
+function mapStateToProps(state, ownProps) {
+    const emojiName = ownProps.emojiName;
+    const customEmojis = getCustomEmojisByName(state);
+
+    let imageUrl = '';
+    let isCustomEmoji = false;
+    let displayTextOnly = false;
+    if (EmojiIndicesByAlias.has(emojiName)) {
+        const emoji = Emojis[EmojiIndicesByAlias.get(emojiName)];
+        imageUrl = Client4.getSystemEmojiImageUrl(emoji.filename);
+    } else if (customEmojis.has(emojiName)) {
+        const emoji = customEmojis.get(emojiName);
+        imageUrl = Client4.getCustomEmojiImageUrl(emoji.id);
```

Comment:
Maybe I had been thinking about this differently before, but weren't we adding a route to get a custom emoji's image by name? This looks like we need to load the emoji definition before we can load the emoji image instead of being able to load them in parallel

Reply:
That route didn't get added. We can chat about it some more if you want but I think trying to do both in parallel might be unneeded complexity. Maybe we can do it without pulling the emoji metadata at all but there's some nuances there that I think we'd have to work through